baybe/utils/chemistry.py

```diff
@@ -72,158 +72,117 @@ def name_to_smiles(name: str) -> str:
+    """Compute molecular fingerprints for a list of SMILES strings.
 
     Args:
-        smiles_list: List of SMILES strings.
+        smiles: Sequence of SMILES strings.
+        encoding: Encoding used to transform SMILES to fingerprints.
         prefix: Name prefix for each descriptor (e.g., nBase --> <prefix>_nBase).
-        dtype: Specifies whether fingerprints will have int or float data type.
-        radius: Radius for the Morgan fingerprint.
-        n_bits:Number of bits for the Morgan fingerprint.
+        kwargs_conformer: kwargs for conformer generator
+        kwargs_fingerprint: kwargs for fingerprint generator
 
     Returns:
-        Dataframe containing Morgan fingerprints for each SMILES string.
+        Dataframe containing fingerprints for each SMILES string.
     """
-    mols = smiles_to_molecules(smiles_list)
+    kwargs_fingerprint = kwargs_fingerprint or {}
+    kwargs_conformer = kwargs_conformer or {}
+
+    if encoding is SubstanceEncoding.MORGAN_FP:
+        warnings.warn(
+            f"Substance encoding '{encoding.name}' is deprecated and will be disabled "
+            f"in a future version. Use '{SubstanceEncoding.ECFP.name}' "
+            f"with 'fp_size' 1204 and 'radius' 4 instead.",
+            DeprecationWarning,
+        )
+        encoding = SubstanceEncoding.ECFP
+        kwargs_fingerprint.update({"fp_size": 1024, "radius": 4})
+
+    elif encoding is SubstanceEncoding.RDKIT:
+        warnings.warn(
+            f"Substance encoding '{encoding.name}' is deprecated and will be disabled "
+            f"in a future version. Use '{SubstanceEncoding.RDKIT2DDESCRIPTORS.name}' "
+            f"instead.",
+            DeprecationWarning,
+        )
+        encoding = SubstanceEncoding.RDKIT2DDESCRIPTORS
 
-    res = []
-    for mol in mols:
-        RDLogger.logger().setLevel(RDLogger.CRITICAL)
+    fingerprint_cls = get_fingerprint_class(encoding)
+    fingerprint_encoder = fingerprint_cls(**kwargs_fingerprint)
 
-        fingerp = GetMorganFingerprintAsBitVect(mol, radius, nBits=n_bits).ToBitString()
-        fingerp = map(int, fingerp)
-        fpvec = np.array(list(fingerp))
-        res.append(
-            {prefix + "FP_" + f"{k + 1}": dtype(bit) for k, bit in enumerate(fpvec)}
+    if fingerprint_encoder.requires_conformers:
+        mol_list = ConformerGenerator(**kwargs_conformer).transform(
+            MolFromSmilesTransformer().transform(smiles)
         )
-
-    df = pd.DataFrame(res)
+    else:
+        mol_list = smiles
+
+    features = np.concatenate(
+        [
+            _molecule_to_fingerprint_features(
+                fingerprint_encoder=fingerprint_encoder, molecule=mol
+            )
+            for mol in mol_list
+        ]
+    )
+    name = f"skfp{fingerprint_encoder.__class__.__name__.removesuffix('Fingerprint')}_"
+    col_names = [prefix + name + f for f in fingerprint_encoder.get_feature_names_out()]
```

Comment:
@Scienfitz any opinion?

Reply:
hmm there are two cases to consider

this is a result of bit-based fingerprint columns being called like that from skfp's side. For those I also see no value in the names and they could simply be called `skfpECFP_123`. We could probably even get rid of the `skfp` OR unify captalizationa s you like (im quite indifferent to that)

However, there are descriptor based fingerprints where the name has meaning like `skfpMORDRED_MolWeight`, we do not want to remove that info --> It might be important for diagnostics purposes. At least that was what I assumed, but a quick check now just showed me that skfp does not seem to provide the proper column name of the descriptor and theya re also just enumerated :/

In that case we could replace the end by just a number BUT i would rather investigate first whether the proper descriptor names are really not accessible (maybe even suggest that as issue in the skfp repo)

Reply:
Sounds good, can one of you open the corresponding issue?
Also, can you specify for these two cases (where the info is available and where it is not available) how exactly the results should look like, in form of an example for each case?

Reply:
[Issue](https://github.com/scikit-fingerprints/scikit-fingerprints/issues/307)

Here is an example for our old column names whichw e could ideally kind of retain:
![image](https://github.com/user-attachments/assets/77596ee5-a6ee-4077-92cf-c5aed4529f19)

For bit fingperprints it was like this:
![image](https://github.com/user-attachments/assets/fb74a456-d8e1-4efb-ae9d-c3f420a8fbd4)

`p` here was just the parameter name i used for the quick test

Reply:
Ok, so the pattern we want is `<parameter_name>_<fingerprint_name>` where `<fingerprint_name>` is either a generic or physical name

Reply:
no Id say `<parameter_name>_<encoding_name>_<feature_name>` where `feature_name` sometimes is just a number resulting from an enumeration and sometimes a meaningful name

According to the issue answer skfp is open in providing this, that would mean we should simply retain the feature names we currently get and the problem will fix itself at some point

Reply:
I think there is prob no easy automatic way of disntinguishing the ones that are bit fps and the ones that are descriptor fps, in that case one culd simply override the long names into numbers for the former case and retain the names for the latter case

Reply:
Ok, I suggest you take care of the necessary change then, since you have the clearer picture in mind than me 🙃 but nice that they responded so quickly

Reply:
Fixed, now it is PREFIX_skfpNAME_FEATURENAME